sys/pre-commit-indent

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+#
+# A hook script to verify what is about to be committed follows the coding
+# style. Called by "git commit" with no arguments. The hook should exit with
+# non-zero status after issuing an appropriate message if it wants to stop the
+# commit.
+#
+# To enable this hook, move it to ".git/hooks/pre-commit".
+
+tmpfile=$(mktemp)
+unstaged_diff=$(mktemp)
+git diff > ${unstaged_diff}
+git checkout -- .
+git diff --cached | ./sys/clang-format-diff.py -p1 > ${tmpfile}
+
```

Comment:
handle ^C to remove the temporary files. see configure script trap_c function to see how to do that